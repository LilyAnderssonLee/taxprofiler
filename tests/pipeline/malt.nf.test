nextflow_pipeline {

    name "Test Workflow main.nf"
    script "main.nf"
    tag "test_nothing"
    tag "malt"
    tag "pipeline"

    test("MALT") {

        when {
            params {
                outdir = "$outputDir"
                input  = "https://github.com/nf-core/test-datasets/raw/taxprofiler/samplesheet_shortreadsonly.csv"
                run_malt = true
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(UTILS.removeNextflowVersion("$outputDir")).match("software_versions") },
                { assert snapshot(path("$outputDir/malt/malt85/malt85-malt-run.log"),
                                path("$outputDir/malt/malt95/malt95-malt-run.log")).match("malt") },
                { assert snapshot(path("$outputDir/multiqc/multiqc_data/malt.txt"),
                                path("$outputDir/multiqc/multiqc_data/multiqc_fastqc.txt"),
                                path("$outputDir/multiqc/multiqc_data/multiqc_general_stats.txt")).match("multiqc") }
            )
        }

    }

}
