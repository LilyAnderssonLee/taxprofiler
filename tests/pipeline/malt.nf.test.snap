{
    "software_versions": {
        "content": [
            "{CUSTOM_DUMPSOFTWAREVERSIONS={python=3.11.0, yaml=6.0}, FASTQC={fastqc=0.11.9}, MALT_RUN={malt=0.6.1}, MEGAN_RMA2INFO_TSV={megan=6.21.7}, SAMPLESHEET_CHECK={python=3.8.3}, UNTAR={untar=1.30}, Workflow={nf-core/taxprofiler=1.1.0dev}}"
        ],
        "timestamp": "2023-06-19T19:53:09+0000"
    },
    "multiqc": {
        "content": [
            "malt.txt:md5,8918575585d4944cb5f50827c709ada6",
            "multiqc_fastqc.txt:md5,cf61413381ea064a476a33f905c0536a",
            "multiqc_general_stats.txt:md5,4f0f2580d9ab15f1e98ff8d7a6b45bf0"
        ],
        "timestamp": "2023-06-19T19:53:09+0000"
    }
}